packages/wallet/ui/src/util/WalletBackendAdapter.js

```diff
@@ -130,15 +130,80 @@ export const makeWalletBridgeFromFollower = (
     ]),
   );
 
+  // We assume just one cosmos purse per brand.
+  const offers = {};
+  const brandToPurse = new Map();
+  const pursePetnameToBrand = new Map();
+
+  const updatePurses = () => {
+    const purses = [];
+    for (const [brand, purse] of brandToPurse.entries()) {
+      if (purse.currentAmount && purse.brandPetname) {
+        pursePetnameToBrand.set(purse.pursePetname, brand);
+        purses.push(purse);
+      }
+    }
+    notifierKits.purses.updater.updateState(harden(purses));
+  };
+
   const followLatest = async () => {
-    for await (const { value: state } of iterateLatest(follower)) {
+    /** @type {number} */
+    let firstHeight;
+    for await (const { blockHeight } of iterateReverse(follower)) {
+      // TODO: Only set firstHeight and break if the value contains all our state.
+      firstHeight = blockHeight;
+    }
+    for await (const { value } of iterateEach(follower, {
+      height: firstHeight,
+    })) {
+      /** @type {import('@agoric/smart-wallet/src/smartWallet').UpdateRecord} */
+      const updateRecord = value;
       if (firstCallback) {
         firstCallback();
+        Object.values(notifierKits).forEach(({ updater }) =>
+          updater.updateState([]),
+        );
         firstCallback = undefined;
       }
-      Object.entries(notifierKits).forEach(([stateName, { updater }]) => {
-        updater.updateState(state[stateName]);
-      });
+      switch (updateRecord.updated) {
+        case 'brand': {
+          const { descriptor } = updateRecord;
+          const purseObj = {
+            ...brandToPurse.get(descriptor.brand),
+            brand: descriptor.brand,
+            brandPetname: descriptor.petname,
+            pursePetname: descriptor.petname,
+            displayInfo: descriptor.displayInfo,
```

Comment:
the intent might be more clear with destructring,
```suggestion
            ...descriptor
```

Reply:
It's a remapping of properties, so I found following the same style was clearest for me.